src/e2e/puppeteer/__tests__/caret-multiline.ts

```diff
@@ -0,0 +1,254 @@
+  })
+
+  it('on cursorUp, the caret should move from the beginning of a cursor to the beginning of the new cursor.', async () => {
+    const importText = `
+    - a
+      - b
+        - c
+        - Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.
+        - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable('b')
+    await click(editableNodeHandle, { edge: 'left' })
+
+    await press('ArrowUp')
+
+    // the focus must be at the beginning of 'a' after cursor up
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe('a')
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBe(0)
+  })
+
+  it('on cursorUp, the caret should move from the end of a cursor to the end of the new cursor.', async () => {
+    const importText = `
+  - a
+    - b
+      - c
+      - Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable('b')
+    await click(editableNodeHandle, { edge: 'right' })
+
+    await press('ArrowUp')
+
+    // the focus must be at the beginning of 'a' after cursor up
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe('a')
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBe(0)
+  })
+
+  it.skip('on 2x cursorDown, the caret should move from the end of a cursor into the 2nd line of the new multi-line cursor.', async () => {
+    const importText = `
+  - a
+    - b
+      - c
+      - Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable('c')
+    await click(editableNodeHandle, { edge: 'left' })
+
+    await press('ArrowDown')
+    await press('ArrowDown')
+
+    // the focus must be in the middle of the multi-line thought after cursor down
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe(
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.",
+    )
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBeGreaterThan(70)
+  })
+
+  it('on cursorDown, the caret should move from the beginning of a multi-line cursor into the 2nd line of the same cursor.', async () => {
```

Comment:
I believe this should start "near the beginning"  (say offset `2`) rather than at the beginning (offset `0`)?

I've numbered the test cases in #1660 to make them easier to refer to. This would correspond to (6) I believe.